backend/typescript/services/implementations/campService.ts

```diff
@@ -85,6 +92,201 @@ class CampService implements ICampService {
+      location: camp.location,
+      fee: camp.fee,
+      formQuestions: oldCamp.formQuestions.map((formQuestion) =>
+        formQuestion.toString(),
+      ),
+    };
+  }
+
+  async createCampSessions(
+    // bulk create
+    campId: string,
+    campSessions: CreateCampSessionsDTO,
+  ): Promise<CampSessionDTO[]> {
+    const insertCampSessions: Omit<CampSessionDTO, "id">[] = [];
+    campSessions.forEach((campSession) => {
+      insertCampSessions.push({
+        camp: campId,
+        campers: [],
+        capacity: campSession.capacity,
+        waitlist: [],
+        dates: campSession.dates.sort(),
+        startTime: campSession.startTime,
+        endTime: campSession.endTime,
+        active: campSession.active,
+      });
+    });
+
+    let newCampSessions: Array<CampSession> = [];
+    let newCampSessionsIds: Array<string>;
+
+    try {
+      newCampSessions = await MgCampSession.insertMany(insertCampSessions);
+      newCampSessionsIds = newCampSessions.map((session) => session.id);
+      await MgCamp.findByIdAndUpdate(
+        campId,
+        {
+          $push: { campSessions: newCampSessionsIds },
+        },
+        { runValidators: true },
+      );
+    } catch (error: unknown) {
+      try {
+        Promise.all(
+          newCampSessions.map(async (session) => {
+            MgCampSession.findByIdAndDelete(session.id);
+          }),
+        );
+      } catch (rollbackError: unknown) {
+        Logger.error(
+          `Failed to rollback camp session creation error. Reason = ${getErrorMessage(
+            rollbackError,
+          )}`,
+        );
+      }
+
+      Logger.error(
+        `Failed to create CampSession. Reason = ${getErrorMessage(error)}`,
+      );
+      throw error;
+    }
+
+    return newCampSessions.map(
+      (session) =>
+        <CampSessionDTO>{
+          id: session.id,
+          camp: campId,
+          campers: [],
+          capacity: session.capacity,
+          waitlist: [],
+          dates: session.dates.map((date) => date.toString()),
+          startTime: session.startTime,
+          endTime: session.endTime,
+          active: session.active,
+        },
+    );
+  }
+
+  async updateCampSessionById(
+    campId: string,
+    campSessionId: string,
+    campSession: UpdateCampSessionDTO,
+  ): Promise<CampSessionDTO> {
+    try {
+      const oldCamp: Camp | null = await MgCamp.findById(campId);
+      if (!oldCamp) {
+        throw new Error(`Camp with campId ${campId} not found.`);
+      }
+      const oldSessions = oldCamp.campSessions as Schema.Types.ObjectId[];
+      if (
+        !oldSessions.find((session) => session.toString() === campSessionId)
+      ) {
+        throw new Error(
+          `CampSession with campSessionId ${campSessionId} not found for Camp with id ${campId}.`,
+        );
+      }
+
+      const newCampSession: CampSession | null = await MgCampSession.findByIdAndUpdate(
+        campSessionId,
+        {
```

Comment:
noticed we can't update capacity, but I think it should be editable as long as the camp hasn't been published. we can make a new task also for updating logic for when we can/can't update fields like fee and capacity on camps/campsessions

Reply:
oops yeah we should def be able to update it here, must have missed this when capacity was moved to session. sounds good!